relay-event-schema/src/protocol/contexts/trace.rs

```diff
@@ -170,6 +173,44 @@ pub struct Route {
     pub other: Object<Value>,
 }
 
+impl FromValue for Route {
+    fn from_value(value: Annotated<Value>) -> Annotated<Self>
+    where
+        Self: Sized,
+    {
+        match value {
+            Annotated(Some(Value::String(name)), meta) => Annotated(
+                Some(Route {
+                    name: Annotated::new(name),
+                    ..Default::default()
+                }),
+                meta,
+            ),
+            Annotated(Some(Value::Object(mut values)), meta) => {
+                let mut route: Route = Default::default();
+                if let Some(Annotated(Some(Value::String(name)), _)) = values.remove("name") {
+                    route.name = Annotated::new(name);
+                }
+                if let Some(Annotated(Some(Value::Object(params)), _)) = values.remove("params") {
+                    route.params = Annotated::new(params);
+                }
+
+                if !values.is_empty() {
+                    route.other = values;
+                }
+
+                Annotated(Some(route), meta)
+            }
+            Annotated(None, meta) => Annotated(None, meta),
+            Annotated(Some(value), mut meta) => {
+                meta.add_error(Error::expected("route"));
```

Comment:
nit (not sure if the message is the most helpful one):
```suggestion
                meta.add_error(Error::expected("route expected to be an object"));
```